### PR TITLE
[deckhouse] fix several deployed module releases

### PR DESCRIPTION
**Pull request by ldmonster**
## Description
It makes outdated deployed module releases superseded at start

## Why do we need it, and what problem does it solve?
In some cases, deployed releases may remain in the deployed status, we need to ensure only one deployed module release, so at restart we make all outdated deployed module releases superseded.

## Why do we need it in the patch release (if we do)?
The module-release-controller runs in the concurrent mode, so several deployed module releases can start race.

## What is the expected result?
Only one module release is deployed.

Several deployed modules releases:

```bash
kubectl get mr -l module=deckhouse-commander
NAME                          PHASE      UPDATE POLICY   TRANSITIONTIME   MESSAGE
deckhouse-commander-v1.3.19   Deployed   deckhouse       47d
deckhouse-commander-v1.3.20   Deployed   deckhouse       47d
deckhouse-commander-v1.4.18   Skipped    deckhouse       47d
deckhouse-commander-v1.4.19   Skipped    deckhouse       47d
deckhouse-commander-v1.4.20   Skipped    deckhouse       47d
deckhouse-commander-v1.4.21   Deployed   deckhouse       47d
```

After restart, only one is deployed:

```bash
kubectl get mr -l module=deckhouse-commander
NAME                          PHASE      UPDATE POLICY   TRANSITIONTIME   MESSAGE
deckhouse-commander-v1.3.19   Superseded   deckhouse       1m
deckhouse-commander-v1.3.20   Superseded   deckhouse       1m
deckhouse-commander-v1.4.18   Skipped    deckhouse       47d
deckhouse-commander-v1.4.19   Skipped    deckhouse       47d
deckhouse-commander-v1.4.20   Skipped    deckhouse       47d
deckhouse-commander-v1.4.21   Deployed   deckhouse       47d
```

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse
type: fix
summary: Fix several deployed module releases.
```

### CODE REVIEW

File: deckhouse-controller/pkg/controller/moduleloader/sync.go [hunk 1]
```diff
@@ -162,18 +165,42 @@ func (l *Loader) restoreAbsentModulesFromOverrides(ctx context.Context) error {
...
+			newRelease := dRelease.DeepCopy()
+			newRelease.Status.Phase = v1alpha1.ModuleReleasePhaseSuperseded
+			newRelease.Status.Message = ""
+			newRelease.Status.TransitionTime = metav1.NewTime(l.dependencyContainer.GetClock().Now().UTC())
+
+			err := l.client.Status().Patch(ctx, newRelease, client.MergeFrom(&dRelease))
+			if err != nil {
+				l.log.Error("patch previous deployed module release", slog.String("name", release.GetName()), log.Err(err))
+			}
+		}
```
Comment (ipaqsa) [inline deckhouse-controller/pkg/controller/moduleloader/sync.go:200]:
```
		// if we already have deployed release - make it superseded
		if deployedRelease, ok := deployedReleases[release.Spec.ModuleName]; ok {
			updatedDeployedRelease := deployedRelease.DeepCopy()
			updatedDeployedRelease.Status.Phase = v1alpha1.ModuleReleasePhaseSuperseded
			updatedDeployedRelease.Status.Message = ""
			updatedDeployedRelease.Status.TransitionTime = metav1.NewTime(l.dependencyContainer.GetClock().Now().UTC())

			if err := l.client.Status().Patch(ctx, updatedDeployedRelease, client.MergeFrom(&deployedRelease)); err != nil {
				l.log.Error("failed to patch previous deployed module release", slog.String("name", release.GetName()), log.Err(err))
			}
		}
```

**Comment by ipaqsa** [comment]
@ldmonster add before/after example in "What is the expected result?" block

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @ldmonster. I've started the workflow for label 'e2e/run/aws'...

**Review by ipaqsa — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [11256](https://github.com/deckhouse/deckhouse/pull/11256) to the branch [release-1.67](https://github.com/deckhouse/deckhouse/tree/release-1.67) successful!

**Comment by deckhouse-BOaTswain** [comment]
Failure: cherry-pick commit be02c17f29e5f2a341af968359e52b51c74cf26e to the branch [release-1.66](https://github.com/deckhouse/deckhouse/tree/release-1.66) failed. See [Job](https://github.com/deckhouse/deckhouse/actions/runs/12493106269) for details.

**Pull request merged**